components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -507,15 +533,22 @@ void ConversationDriver::GenerateQuestions() {
   suggestion_generation_status_ =
       mojom::SuggestionGenerationStatus::IsGenerating;
   OnSuggestedQuestionsChanged();
-  // Make API request for questions.
+  // Make API request for questions but first get page content.
   // Do not call SetRequestInProgress, this progress
   // does not need to be shown to the UI.
-  auto navigation_id_for_query = current_navigation_id_;
-  engine_->GenerateQuestionSuggestions(
-      is_video_, article_text_,
-      base::BindOnce(&ConversationDriver::OnSuggestedQuestionsResponse,
-                     weak_ptr_factory_.GetWeakPtr(),
-                     std::move(navigation_id_for_query)));
+  auto on_content_retrieved = [](ConversationDriver* instance,
+                                 int64_t navigation_id,
+                                 std::string page_content, bool is_video,
+                                 std::string invalidation_token) {
+    instance->engine_->GenerateQuestionSuggestions(
+        is_video, page_content,
+        base::BindOnce(&ConversationDriver::OnSuggestedQuestionsResponse,
+                       instance->weak_ptr_factory_.GetWeakPtr(),
+                       std::move(navigation_id)));
+  };
+  GeneratePageContent(base::BindOnce(std::move(on_content_retrieved),
+                                     base::Unretained(this),
```

Comment:
This is safe because the callback is passed to a function in this class (itself via a weakptr), so if the class instance is destroyed then that callback wouldn't get called.